constants.gradle

```diff
@@ -0,0 +1,62 @@
+ext {
+    // build dependencies
+    gradleBuildToolsVersion = '7.3.1'
+    gradleKotlinVersion = '1.7.20'
+    dokkaVersion = '1.7.20'
+    gradleKtlintVersion = "11.2.0"
+
+    // android versioning
+    androidCompileVersion = 33
+    buildToolsVersion = '30.0.3'
+    minSDKVersion = 23
+    targetSDKVersion = 33
+
+    // project naming and versioning
+    groupId = 'com.klaviyo'
+    versionCode = 1
+    versionName = '1.0.1'
+
+    // dependencies
+    coreKTXVersion = '1.9.0'
+    kotlinSTDVersion = '1.7.20'
+    fcmVersion = '23.1.1'
+
+    // sample app dependencies
+    composeUiVersion = '1.3.3'
+    materialVersion = '1.3.1'
+    googleServicesVersion = '4.3.15'
+
+    // testing dependencies
+    junitVersion = '4.13.2'
+    junitAndroidVersion = '1.1.5'
+    testRunnerVersion = '1.5.2'
+    testRulesVersion = '1.5.0'
+    mockkVersion = "1.13.3"
+    jsonVersion = '20200518'
+
+    // instrumentation testing dependencies
+    espressoVersion = '3.5.1'
+
+    /**
+     * Extracted product flavor as a closure to be applied for all android projects
+     */
+    flavorConfig = {
+        def localProperties = new Properties()
+        if (rootProject.file("local.properties").canRead()) {
+            localProperties.load(new FileInputStream(rootProject.file("local.properties")))
+        }
+
+        flavorDimensions "server"
```

Comment:
Hmm I forget what these mean but does it make sense for flavor dimension and dimension to be the same?

Reply:
I think this line is how you specify all the available dimensions for flavors, and below is how you specify which dimension  _this_ flavor is for. Just so happens I only have 1 flavor dimension so it looks repetitive.

Reply:
Like we could theoretically have build types of `debug` and `release`, flavors defined by `flavorDimensions "server", "ice_cream"` and then ultimately have build _variants_ like this: 
```
productionVanillaRelease
productionChocolateRelease
debugVanillaRelease
...etc
```